src/compute/src/extensions/arrange.rs

```diff
@@ -211,6 +211,11 @@ where
     let mut trace = arranged.trace.clone();
     let operator = trace.operator().global_id;
 
+    // We don't want to block compaction.
+    let empty = Antichain::new();
+    trace.set_logical_compaction(empty.borrow());
+    trace.set_physical_compaction(empty.borrow());
```

Comment:
I remember you saying that this doesn't work to unblock compaction, which is why you added the manual advancement by the upper in the first place. Did that change?

Reply:
It does not work if there is only a single holder of the trace. Since we now make it the user's responsibility to advance compaction, it is safe to only advance once here.

Reply:
I see, thanks!